ydb/core/tx/datashard/ut_common/datashard_ut_common.cpp

```diff
@@ -2515,4 +2556,123 @@ void SendViaPipeCache(
         /* viaActorSystem */ true);
 }
 
+void AddKeyQuery(
+    TEvDataShard::TEvRead& request,
+    const std::vector<ui32>& keys)
+{
+    // convertion is ugly, but for tests is OK
+    auto cells = ToCells(keys);
+    request.Keys.emplace_back(cells);
+}
+
+void AddFullRangeQuery(TEvDataShard::TEvRead& request) {
+    auto fromBuf = TSerializedCellVec::Serialize(TVector<TCell>());
+    auto toBuf = TSerializedCellVec::Serialize(TVector<TCell>());
+    request.Ranges.emplace_back(fromBuf, toBuf, true, true);
+}
+
+std::unique_ptr<TEvDataShard::TEvRead> GetBaseReadRequest(
+    const TTableId& tableId,
+    const NKikimrTxDataShard::TEvGetInfoResponse::TUserTable& userTable,
+    ui64 readId,
+    NKikimrDataEvents::EDataFormat format,
+    const TRowVersion& readVersion)
+{
+    auto request = std::make_unique<TEvDataShard::TEvRead>();
+    auto& record = request->Record;
+
+    record.SetReadId(readId);
+    record.MutableTableId()->SetOwnerId(tableId.PathId.OwnerId);
+    record.MutableTableId()->SetTableId(userTable.GetPathId());
+
+    const auto& description = userTable.GetDescription();
+    for (const auto& column: description.GetColumns()) {
+        record.AddColumns(column.GetId());
+    }
+
+    record.MutableTableId()->SetSchemaVersion(description.GetTableSchemaVersion());
+
+    if (readVersion) {
+        record.MutableSnapshot()->SetStep(readVersion.Step);
+        record.MutableSnapshot()->SetTxId(readVersion.TxId);
+    }
+
+    record.SetResultFormat(format);
+
+    return request;
+}
+
+std::unique_ptr<TEvDataShard::TEvReadResult> WaitReadResult(Tests::TServer::TPtr server, TDuration timeout) {
+    auto& runtime = *server->GetRuntime();
+    TAutoPtr<IEventHandle> handle;
+    runtime.GrabEdgeEventRethrow<TEvDataShard::TEvReadResult>(handle, timeout);
+    if (!handle) {
+        return nullptr;
+    }
+    std::unique_ptr<TEvDataShard::TEvReadResult> event(handle->Release<TEvDataShard::TEvReadResult>().Release());
+    return event;
+}
+
+void SendReadAsync(
+    Tests::TServer::TPtr server,
+    ui64 tabletId,
+    TEvDataShard::TEvRead* request,
+    TActorId sender,
+    ui32 node,
+    const NTabletPipe::TClientConfig& clientConfig,
+    TActorId clientId)
+{
+    auto &runtime = *server->GetRuntime();
+    runtime.SendToPipe(
+        tabletId,
+        sender,
+        request,
+        node,
+        clientConfig,
+        clientId);
+}
+
+std::unique_ptr<TEvDataShard::TEvReadResult> SendRead(
+    Tests::TServer::TPtr server,
+    ui64 tabletId,
+    TEvDataShard::TEvRead* request,
+    TActorId sender,
+    ui32 node,
+    const NTabletPipe::TClientConfig& clientConfig,
+    TActorId clientId,
+    TDuration timeout)
+{
+    SendReadAsync(server, tabletId, request, sender, node, clientConfig, clientId);
+
+    return WaitReadResult(server, timeout);
+}
+
+TString ReadTable(
+    Tests::TServer::TPtr server,
+    std::span<const ui64> tabletIds,
+    const TString& tableName,
```

Comment:
```suggestion
```

`tableName` should be removed due to
https://github.com/ydb-platform/ydb/pull/7121/files#r1693924255

Reply:
We still need `userTable` for column Ids (for making requests) and for column names and column types (for printing).

Reply:
> We still need `userTable` for column Ids (for making requests) and for column names and column types (for printing).

Yes, you are right.

Reply:
Nevertheless, let's try remove `tableName`. You can enumerate `tablesMap` and find the `userTable` by `tableId.PathId`. 

If you can do it then the result function call would trivial like that:
```
auto tableState = ReadTable(server, shards, tableId);
```